hardware_applications/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.6.2)
+project(hardware_applications)
+
+set(CMAKE_CXX_STANDARD 14)
+
+include_directories(STSL/include)
+
+if(WIN32)
+    link_directories(STSL/lib/WINDOWS)
+    configure_file(STSL/lib/WINDOWS/libSTSL.dll ${CMAKE_CURRENT_BINARY_DIR}/libSTSL.dll COPYONLY)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+elseif(UNIX AND NOT APPLE)
+    link_directories(STSL/lib/LINUX)
+    set(CMAKE_CXX_STANDARD_LIBRARIES /usr/lib/x86_64-linux-gnu/libudev.so)
```

Comment:
hardcoding the paths such as `libudev.so` is really bad practice and makes it so everyone who's not on your specific distribution experiences breakage. The locations of paths can even change between versions of the same operating system or versions of the library. For example, it's in `/lib/x86_64-linux-gnu/libudev.so` on my system.

Maybe you could try using [one of the many FindUdev files online](https://duckduckgo.com/?q=findudev+cmake+site%3Agithub.com&kc=1&k8=a6aaab&k1=1&kx=909396&ky=2e2e2e&kaa=B68800&kp=1&k9=ce4139&kj=262626&kam=osm&k5=1&kaj=m&kn=-1&k7=262626&kao=1&kae=d&kv=1&kak=-1&ia=web).

Reply:
Yeah I know
Thought I'd fixed this already, but it got lost in all the stuff that happened this week.
Rebooting into Linux to fix this now